daemons/controld/controld_cib.c

```diff
@@ -401,57 +401,85 @@ cib_delete_callback(xmlNode *msg, int call_id, int rc, xmlNode *output,
 
 /*!
  * \internal
- * \brief Delete subsection of a node's CIB node_state
+ * \brief Get the XPath and description of a node state section to be deleted
  *
- * \param[in] uname    Desired node
- * \param[in] section  Subsection of node_state to delete
- * \param[in] options  CIB call options to use
+ * \param[in]  uname    Desired node
+ * \param[in]  section  Subsection of node_state to be deleted
+ * \param[out] xpath    Where to store XPath of \p section
+ * \param[out] desc     If not \c NULL, where to store description of \p section
```

Comment:
Document that the contents of xpath and desc will be freed if not NULL (which means they must be valid in any case)

Reply:
Will do. Let me know if you have a different preference on freeing vs. not freeing the contents of a pointer-to-pointer argument BTW.

This will probably cause a segfault if the value is uninitialized, but our standards are to initialize to `NULL`. I prefer to follow that and avoid memory leaks.

Reply:
It's shorter NOT to free them though, and trust the caller

Reply:
come to think of it we rarely free an output argument anywhere in existing code...

Reply:
True -- either make it in,out or don't free it. Either way is fine as long as it's documented.